propagation-stackdriver/src/main/java/zipkin2/propagation/stackdriver/XCloudTraceContextExtractor.java

```diff
@@ -50,37 +51,43 @@
     if (xCloudTraceContext != null) {
       String[] tokens = xCloudTraceContext.split("/");
 
-      // Try to parse the trace IDs into the context
-      TraceContext.Builder context = TraceContext.newBuilder();
       long[] traceId = convertHexTraceIdToLong(tokens[0]);
 
       // traceId is null if invalid
       if (traceId != null) {
-        String spanId = "1";
+        long spanId = 0; // 0 indicates no span ID is set by the user
         Boolean traceTrue = null; // null means to defer trace decision to sampler
```

Comment:
Not part of this change but it might be a good idea to replace the `Boolean` to primitive `boolean` with this.

Reply:
I thought `Boolean` here was necessary because when this is passed into the traceContext builder it has the following meanings:

- `true`: force request to be traced
- `false`: force request not to be traced
- `null`: Let the sampler decide on whether it should be traced or not.

Reply:
incidentally, the `TraceIdContext` type should be used when you have no span ID. that's probably better than using the incoming trace ID, as it is more to the point.

Reply:
@adriancole Right, in this PR I am using the TraceIdContext when I have no span ID. Or am I misunderstanding your comment?

Reply:
you got it.